plugins/git4idea/src/git4idea/repo/GitRepositoryReader.java

```diff
@@ -253,6 +257,12 @@ else if (refName.startsWith(REFS_REMOTES_PREFIX)) {
     return Pair.create(localBranches, remoteBranches);
   }
 
+  private static GitLocalBranch fromRefInfo(RefInfo refInfo) {
+    return refInfo.isSymbolicRef ?
+           new GitSymbolicRef(refInfo.refName, new GitLocalBranch(refInfo.referencedRef)) :
```

Comment:
Here a new `GitLocalBranch` is created instead of reusing existing one. I guess this is not a problem though, since these GitLocalBranch objects are equal, but it would be more clear to reuse the existing branch object, e.g. first by creating objects for non symrefs, and then for symrefs.